### PR TITLE
Add extra rule to check if tour ends.

### PR DESCRIPTION
**Pull request by sashadoes**
## Summary

Addresses issue:

- #6478 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/feature-tours/dashboard-sharing.js [hunk 1]
```diff
@@ -68,7 +68,8 @@ const dashboardSharing = {
 		if (
 			ACTIONS.STOP === action ||
 			ACTIONS.CLOSE === action ||
-			( index === 0 && dialogOpen )
+			( index === 0 && dialogOpen ) ||
+			index + 1 > size
```
Comment (aaemnnosttv) [inline assets/js/feature-tours/dashboard-sharing.js:72]: This doesn't seem to address the issue as `index + 1` would at most equal the `size`.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @sashadoes but unfortunately this doesn't fix the issue.

The close action only handles the X button on the tour, but tours can also be closed by going to the last step and clicking "Got it!" which I think is your intention here.

From what I can see, we need to check for action being `next` and lifecycle being `complete`.

File: assets/js/feature-tours/dashboard-sharing.js (hunk 1 above)
Comment (sashadoes) [reply assets/js/feature-tours/dashboard-sharing.js:72]: A debugged and there is a case when the index gets 4 on the last step, where we should actually close the modal.
 It should help.

**Review by tofumatt — APPROVED** [review]

**Review by tofumatt — CHANGES_REQUESTED** [review]
I'm seeing a React Joyride error, I'm not sure if caused by this change or not, looking into it…

CleanShot 2023-03-09 at 11 25 54

Edit: it seems unrelated, so we won't worry about it in this PR.

**Review by tofumatt — APPROVED** [review]
This fixes the issue mentioned, but I think we need a follow-up issue for the generally buggy behaviour of this feature tour.

I've filed: #6703 for these issues.

**Pull request merged**